### PR TITLE
Devirtualize numel()

### LINKED ISSUE #27291: Devirtualize numel
In #27274 @ljk53 observed that numel is being called a lot. To me, this was quite surprising, as I expected numel to be devirtualized. But it's in `native_functions.yaml`, which means it is not devirtualized. We should devirtualize it.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27294 Devirtualize numel()**

Fixes #27291

I'm a little annoyed that I have to reintroduce manual binding code.  But it's
probably not a good idea to teach the codegen how to do fastpath functions
(is it?)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D17763486](https://our.internmc.facebook.com/intern/diff/D17763486)

### CODE REVIEW

**Comment by ezyang** [comment]
@ailzhang this one breaks xla but it should be a pretty simple fix; just delete the numel() implementation. (This is under the assumption that your TensorImpl::numel implementation does the right thing. Which it should.)

**Comment by ailzhang** [comment]
Thanks @ezyang , I'm starting a local run to verify the change. :D Will report back once it's finished.

File: tools/jit/gen_jit_dispatch.py [hunk 1]
```diff
@@ -369,14 +369,15 @@ def declkey(decl):
         return [sorted(g, key=declkey) for g in grouped_decls]
 
     # We need to add methods implemented manually in TensorImpl
+    # TODO: This seems to claim sizes() returns an int64_t.  Really?
```
Comment (ailzhang) [inline tools/jit/gen_jit_dispatch.py]: our `Tensor::sizes()` does return a int64_t iirc. It's number of dimensions. And yes I agree it's confusing :P

**Review by ailzhang — APPROVED** [review]
Fix is in https://github.com/pytorch/xla/pull/1136

**Pull request closed without merging**